pkg/cmd/util/clientcmd/errors.go

```diff
@@ -68,6 +81,8 @@ func detectReason(err error) int {
 			return noServerFoundReason
 		case clientcmd.IsConfigurationInvalid(err):
 			return configurationInvalidReason
+		case strings.Contains(err.Error(), "tls: oversized record received"):
+			return tlsOversizedRecordReason
```

Comment:
add a unit test that starts up a httptest server, attempts to dial it without TLS, and hits this detection, to make sure we catch any message changes with different go versions